qiita_db/user.py

```diff
@@ -509,6 +530,81 @@ def _change_pass(self, newpass):
             TRN.add(sql, [hash_password(newpass), self._id])
             TRN.execute()
 
+    def messages(self, count=None):
+        """Return messages in user's queue
+
+        Parameters
+        ----------
+        count : int, optional
+            Number of messages to return, starting with newest. Default all
+
+        Returns
+        -------
+        list of tuples
+            Messages in the queue, in the form
+            [(msg_id, msg, timestamp, read, system_message), ...]
+
+        Notes
+        -----
+        system_message is a bool. When True, this is a systemwide message.
+        """
+        with TRN:
+            sql_info = [self._id]
+            sql = """SELECT message_id, message, message_time, read,
+                     (expiration IS NOT NULL) as system_message
+                     FROM qiita.message_user
+                     JOIN qiita.message USING (message_id)
+                     WHERE email = %s ORDER BY message_time DESC"""
+            if count is not None:
+                sql += " LIMIT %s"
+                sql_info.append(count)
+            TRN.add(sql, sql_info)
+            return TRN.execute_fetchindex()
+
+    def mark_messages(self, messages, read=True):
+        """Mark given messages as read/unread
+
+        Parameters
+        ----------
+        messages : list of ints
+            Message IDs to mark as read/unread
+        read : bool, optional
+            Marks as read if True, unread if False. Default True
+        """
+        with TRN:
+            sql = """UPDATE qiita.message_user
+                     SET read = %s
+                     WHERE message_id IN %s AND email = %s"""
+            TRN.add(sql, [read, tuple(messages), self._id])
+            return TRN.execute_fetchindex()
+
+    def delete_messages(self, messages):
+        """Delete given messages for the user
+
+        Parameters
+        ----------
+        messages : list of ints
+            Message IDs to delete
+        """
+        with TRN:
+            # remove message from user
+            sql = """DELETE FROM qiita.message_user
+                     WHERE message_id IN %s AND email = %s
+                     AND message_id NOT IN
```

Comment:
Add an extra indentation here and the next 2 lines, for SQL guidelines compliance